File: python/tests/test_array_pair_sum.py
import unittest
from array_pair_sum import two_sum, two_sum_w_dict

class TestArrayPairSum(unittest.TestCase):
    def test_two_sum(self):
        output_to_test = two_sum([1, 2, 3, 4, 5], 5)
        self.assertCountEqual([[4, 1], [3, 2]], output_to_test)

    def test_two_sum_w_dict(self):
        output_to_test = two_sum_w_dict([1, 2, 3, 4, 5], 5)
        self.assertCountEqual([[4, 1], [3, 2]], output_to_test)
    
if __name__ == '__main__':
    unittest.main()

File: ruby/tests/test_array_pair_sum.rb
require 'minitest/autorun'
require_relative '../array_pair_sum'

class TestArrayPairSum < Minitest::Test
  def test_two_sum
    output_to_test = two_sum([1, 2, 3, 4, 5], 5)
    output_to_test.each do |arr|
      arr.sort!
    end
    output_to_test.sort!

    assert_equal [[1, 4], [2, 3]], output_to_test
  end

  def test_two_sum_w_hash
    output_to_test = two_sum_w_hash([1, 2, 3, 4, 5], 5)
    output_to_test.each do |arr|
      arr.sort!
    end
    output_to_test.sort!

    assert_equal [[1, 4], [2, 3]], output_to_test
  end
end